src/main/java/net/dv8tion/jda/api/utils/messages/MessagePollBuilder.java

```diff
@@ -0,0 +1,271 @@
+     *
+     * @param  layout
+     *         The layout
+     *
+     * @throws IllegalArgumentException
+     *         If null or {@link net.dv8tion.jda.api.entities.messages.MessagePoll.LayoutType#UNKNOWN UNKNOWN} is provided
+     *
+     * @return The updated builder
+     */
+    @Nonnull
+    public MessagePollBuilder setLayout(@Nonnull MessagePoll.LayoutType layout)
+    {
+        Checks.notNull(layout, "Layout");
+        Checks.check(layout != MessagePoll.LayoutType.UNKNOWN, "Layout cannot be UNKNOWN");
+
+        this.layout = layout;
+        return this;
+    }
+
+    /**
+     * Change the title for this poll.
+     *
+     * @param  title
+     *         The poll title (up to {@link MessagePoll#MAX_QUESTION_TEXT_LENGTH} characters)
+     *
+     * @throws IllegalArgumentException
+     *         If the title is blank or longer than {@link MessagePoll#MAX_QUESTION_TEXT_LENGTH} characters
+     *
+     * @return The updated builder
+     */
+    @Nonnull
+    public MessagePollBuilder setTitle(@Nonnull String title)
+    {
+        Checks.notBlank(title, "Title");
+        title = title.trim();
+        Checks.notLonger(title, MessagePoll.MAX_QUESTION_TEXT_LENGTH, "Poll question title");
+
+        this.title = title;
+        return this;
+    }
+
+    /**
+     * Change the duration for this poll.
+     * <br>Default: {@code 1} day
+     *
+     * <p>The poll will automatically expire after this duration.
+     *
+     * @param  duration
+     *         The duration of this poll (in hours resolution)
+     *
+     * @throws IllegalArgumentException
+     *         If the duration is null, less than 1 hour, or longer than {@value  MessagePoll#MAX_DURATION_HOURS} hours (7 days)
+     *
+     * @return The updated builder
+     */
+    @Nonnull
+    public MessagePollBuilder setDuration(@Nonnull Duration duration)
+    {
+        Checks.notNull(duration, "Duration");
+        Checks.positive(duration.toHours(), "Duration");
+        Checks.check(duration.toHours() <= MessagePoll.MAX_DURATION_HOURS, "Poll duration may not be longer than 7 days. Provided: %d hours", duration.toHours());
+
+        this.duration = duration;
+        return this;
+    }
+
+    /**
+     * Change the duration for this poll.
+     * <br>Default: {@code 1} day
+     *
+     * <p>The poll will automatically expire after this duration.
+     *
+     * @param  duration
+     *         The duration of this poll (in hours resolution)
+     * @param  unit
+     *         The time unit for the duration
+     *
+     * @throws IllegalArgumentException
+     *         If the time unit is null or the duration is not between 1 and {@value  MessagePoll#MAX_DURATION_HOURS} hours (7 days) long
+     *
+     * @return The updated builder
+     */
+    @Nonnull
+    public MessagePollBuilder setDuration(long duration, @Nonnull TimeUnit unit)
+    {
+        Checks.notNull(unit, "TimeUnit");
+        this.duration = Duration.ofHours(unit.toHours(duration));
```

Comment:
Shouldn't there be a check for max duration here to ensure it doesn't go beyond the total?

Reply:
This was supposed to call the other setter, good catch